### DISCUSSION TITLE
HelmRelease and Kustomization

### DISCUSSION THREAD

**Discussion by Thaval** [fluxcd/flux2#2722] [Q&A]
Hi, 

how does the combination of HelmRelease and Kustomization work? I'm trying to understand the concept of both together.
Imagine I create the HelmRelease first, then create a kustomization.yaml and then the Kustomization component. So the HelmRelease will create the HelmChart and will deploy the workloads by the rendering the HelmChart with the provided values.yaml(s). Then Kustomization will detect that there's a HelmRelease (because the resource is assigned to it) and will try to applied the HelmRelease. 
So.. the HelmRelease is to initialize and the Kustomization is to synchronize the HelmRelease with the kubernetes cluster, right? 
So the only purpose of Kustomization is to actually watch for HelmRelease changes and apply these changes? This would be mean that the Kustomization uses kinda the same logic and commands that HelmRelease uses, correct?

**Comment by kingdonb** [comment]
The Kustomization applies a HelmRelease to the cluster, so it manages the metadata and the spec. This is the core through which everything gets installed by Flux on your cluster.

The HelmRelease is what actually manages running "Helm" activities through the Helm package SDK.

A good overview of how to use these components together is here:

https://github.com/fluxcd/flux2-kustomize-helm-example

You might find a better understanding of the basic resources that Flux uses to manage workloads in your cluster by starting here:

https://fluxcd.io/docs/concepts/
